torch/utils/_sympy/functions.py

```diff
@@ -761,49 +762,30 @@ def _find_localzeros(cls, values, **options):
         When a value is identified as being more extreme than another member it
         replaces that member; if this is never true, then the value is simply
         appended to the localzeros.
-        """
-        localzeros = set()  # type: ignore[var-annotated]
-        for v in values:
-            is_newzero = True
-            localzeros_ = list(localzeros)
-            for z in localzeros_:
-                if id(v) == id(z):
-                    is_newzero = False
-                else:
-                    con = cls._is_connected(v, z)
-                    if con:
-                        is_newzero = False
-                        if con is True or con == cls:
-                            localzeros.remove(z)
-                            localzeros.update([v])
-            if is_newzero:
-                localzeros.update([v])
-        return localzeros
 
-    @classmethod
-    def _is_connected(cls, x, y):
-        """
-        Check if x and y are connected somehow.
+        Unlike the sympy implementation, we only look for zero and one, we don't
+        do generic is connected test pairwise which is slow
         """
-        if x == y:
-            return True
-        t, f = Max, Min
-        for op in "><":
-            for j in range(2):
-                try:
-                    if op == ">":
-                        v = x >= y
-                    else:
-                        v = x <= y
-                except TypeError:
-                    return False  # non-real arg
-                if not v.is_Relational:
-                    return t if v else f
-                t, f = f, t  # type: ignore[assignment]
-                x, y = y, x
-            x, y = y, x  # run next pass with reversed order relative to start
-
-        return False
+        if len(values) == 2:
+            ix = 1 if cls is Max else 0
+            if values[0] in (0.0, 0) and values[1].is_nonnegative:
+                return {values[ix]}
+            elif values[1] in (0.0, 0) and values[0].is_nonnegative:
+                return {values[1 - ix]}
+            elif values[1] == 1 and values[0].is_positive:
+                return {values[1 - ix]}
```

Comment:
Return 1 in case `values[0].is_nonpositive`?

Reply:
Do you think maybe I should just use the old is connected algorithm for the 2-ary case? Maybe that will also be fast.

Reply:
I don't think it would be fast. What you have here is pretty fast.

Reply:
okey dokey. In that case, I'm going to omit the nonpositive case, we basically never have values like that lol